easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -563,22 +551,85 @@ def include_defined_parameters(keyset):
         ebtxt.extend(self.comments['header'])
 
         # print easyconfig parameters ordered and in groups specified above
-        include_defined_parameters(GROUPED_PARAMS)
+        include_defined_parameters(ebtxt, GROUPED_PARAMS)
 
         # print other easyconfig parameters at the end
         keys_to_ignore = printed_keys + LAST_PARAMS
         for key in default_values:
             if key not in keys_to_ignore and self[key] != default_values[key]:
-                add_key_and_comments(key, quote_py_str(self[key]))
+                ebtxt = self._add_key_and_comments(ebtxt, key, quote_py_str(self[key]), templ_const, templ_val, formatting)
         ebtxt.append('')
 
         # print last parameters
-        include_defined_parameters([[k] for k in LAST_PARAMS])
+        include_defined_parameters(ebtxt, [[k] for k in LAST_PARAMS])
 
         write_file(fp, ('\n'.join(ebtxt)).strip()) # strip for newlines at the end
 
+        dumped_text = ('\n'.join(ebtxt))
+        write_file(fp, (fix_code(dumped_text, options={'aggressive': 1, 'max_line_length':120})).strip())
         self.enable_templating = orig_enable_templating
 
+    def _add_key_and_comments(self, ebtxt, key, val, templ_const, templ_val, formatting):
+        """ Add key, value pair and comments (if there are any) to the dump file (helper method for dump()) """
+        if formatting:
+            val = self._format(key, val, True, dict())
+        else:
+            val = str(val)
+
+        # templates
+        if key not in EXCLUDED_KEYS_REPLACE_TEMPLATES:
+            new_val = to_template_str(val, templ_const, templ_val)
+            if not r'%(' + key in new_val:
+                val = new_val
+
+        if key in self.comments['inline']:
+            ebtxt.append("%s = %s%s" % (key, val, self.comments['inline'][key]))
+        else:
+            if key in self.comments['above']:
+                ebtxt.extend(self.comments['above'][key])
+            ebtxt.append("%s = %s" % (key, val))
+
+        return ebtxt
```

Comment:
rather than appending to the variable you're being passed (eek!), just return the lines to be added?